From 9a2e03bb6b098776d2008af49d22b912f410dd2d Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Thu, 7 Jun 2018 15:57:23 +0300
Subject: [PATCH 3945/5242] MLK-16784 sound: asoc: change micfil dma_maxburst

commit  2a52c20b055fac4f276a0b99837fc4172fd32ab7 from
https://source.codeaurora.org/external/imx/linux-imx.git

MICFIL is using multi-fifo dma and maxburst must
be set proportional to number of channels.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index d108f33..47953f0 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -405,6 +405,7 @@ static int fsl_micfil_hw_params(struct snd_pcm_substream *substream,
 	}
 
 	micfil->dma_params_rx.fifo_num = channels;
+	micfil->dma_params_rx.maxburst = channels * MICFIL_DMA_MAXBURST_RX;
 
 	return 0;
 }
-- 
1.7.9.5

